packages/plugin-table/src/TableCellNode.ts

```diff
@@ -0,0 +1,159 @@
+    __managedCells = new Set<TableCellNode>();
+
+    constructor(public header = false) {
+        super();
+    }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     *  @override
+     */
+    clone(): this {
+        const clone = new this.constructor<typeof TableCellNode>(this.header);
+        clone.attributes = { ...this.attributes };
+        return clone;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getters
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    get name(): string {
+        let coordinatesRepr = ' <(' + this.rowIndex + ', ' + this.columnIndex + ')';
+        if (this.colspan > 1 || this.rowspan > 1) {
+            const endRow = this.rowIndex + this.rowspan - 1;
+            const endColumn = this.columnIndex + this.colspan - 1;
+            coordinatesRepr += ':(' + endRow + ', ' + endColumn + ')';
+        }
+        coordinatesRepr += '>';
+        return (
+            super.name +
+            coordinatesRepr +
+            (this.header ? ': header' : '') +
+            (this.isActive ? '' : ' (inactive)')
+        );
+    }
+    /**
+     * Return the cell that manages this cell, if any.
+     */
+    get managerCell(): TableCellNode {
+        return this.__managerCell;
+    }
+    /**
+     * Set the given cell as manager of this cell. If null is passed, this cell
+     * will have no manager.
+     */
+    set managerCell(newManager: TableCellNode | null) {
+        const oldManager = this.__managerCell;
+        this.__managerCell = newManager;
+
+        if (newManager) {
+            // If we just set a new manager for this cell, also add this cell to
+            // the new manager's managed cells.
+            newManager.__managedCells.add(this);
+
+            // Copy the manager.
+            this.attributes = { ...newManager.attributes };
+            this.header = newManager.header;
+
+            // Move the children to the manager.
+            newManager.append(...this.children);
+
+            // Hand the managed cells over to the manager.
+            for (const managedCell of this.managedCells) {
+                managedCell.managerCell = newManager;
+                this.__managedCells.delete(managedCell);
+            }
+        } else if (oldManager) {
+            // If we just removed this cell's manager, also remove this cell
+            // from the old manager's managed cells.
+            oldManager.__managedCells.delete(this);
+        }
+    }
+    /**
+     * Return the set of cells that this cell manages.
+     */
+    get managedCells(): Set<TableCellNode> {
+        return new Set(this.__managedCells);
```

Comment:
Because of this getter and the hiding of the property, you are accessing the private property from the outside here: https://github.com/odoo-dev/jabberwock/pull/106/commits/c50eafebc0093984ccd685931b70485e0ddda044#diff-8696de8087e2f8c00c3b0a80ad497247R74

Reply:
Well, if you do `<anything_other_than_"this">.__<private_property>` then yes, for me it is the outside. I'm wondering if it would be better or worse if instead of doing:
```typescript
managedCell.managerCell = manager;
```

We did:
```typescript
managerCell.manage(managedCell);
```

But I guess we're a bit screwed on that since we decided to use `mergeWith` as it would also need to write on the manager's private property from the outside...

Reply:
Exactly...

You:
> if you do <anything_other_than_"this">.__<private_property> then yes, for me it is the outside

Code:
> `this.__managedCells`

Problem:
> ...

Reply:
Re-read my comment ! 🚎  The problem is not here, it is there: c50eafe#diff-8696de8087e2f8c00c3b0a80ad497247R74
I put the comment here because it is the existence of this getter that is somewhat the *cause* of the issue... ;-)

Reply:
I think we need to discuss these private properties by audio. I still need to work with christophe tomorrow but maybe we can squeeze like 10 minutes to discuss this.